### ISSUE TITLE
Shuffling rows isn't working correctly

### ISSUE THREAD

**Issue by FreshDonut** [yconst/YCMatrix#3]
When I try shuffling rows, it messes up by putting values in different columns. I want to shuffle rows, but keep every value within their respective columns.

Here's an example.

Unshuffled:

``` objective-c
1.000000    10.000000   100.000000  1000.000000 10000.000000
2.000000    20.000000   200.000000  2000.000000 20000.000000
3.000000    30.000000   300.000000  3000.000000 30000.000000
4.000000    40.000000   400.000000  4000.000000 40000.000000
```

Shuffled (Both in place shuffling and out-of-place):

``` objective-c
300.000000  3000.000000 30000.000000    4.000000    40.000000
10.000000   100.000000  1000.000000 1.000000    2.000000
20.000000   200.000000  10000.000000    20000.000000    3.000000
30.000000   2000.000000 400.000000  4000.000000 40000.000000
```

Expected Result:

``` objective-c
4.000000    40.000000   400.000000  4000.000000 40000.000000
1.000000    10.000000   100.000000  1000.000000 10000.000000
3.000000    30.000000   300.000000  3000.000000 30000.000000
2.000000    20.000000   200.000000  2000.000000 20000.000000
```

**Comment by yconst** [comment]
Thanks I'll look into it.

**Comment by yconst** [comment]
This has been fixed (& relevant tests added) as of 0.6.3. Thanks for reporting.
